sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -457,16 +457,16 @@ public void init(CosmosClientMetadataCachesSnapshot metadataCachesSnapshot, Func
                 collectionCache);
 
             updateGatewayProxy();
-            if (this.connectionPolicy.getConnectionMode() == ConnectionMode.GATEWAY) {
-                this.storeModel = this.gatewayProxy;
-            } else {
-                this.initializeDirectConnectivity();
-            }
             clientTelemetry = new ClientTelemetry(null, UUID.randomUUID().toString(),
                 ManagementFactory.getRuntimeMXBean().getName(), userAgentContainer.getUserAgent(),
                 connectionPolicy.getConnectionMode(), globalEndpointManager.getLatestDatabaseAccount().getId(),
                 null, null, this.reactorHttpClient, connectionPolicy.isClientTelemetryEnabled(), this, this.connectionPolicy.getPreferredRegions());
             clientTelemetry.init();
+            if (this.connectionPolicy.getConnectionMode() == ConnectionMode.GATEWAY) {
```

Comment:
why are we changing this order or direct / gateway mode initialization?

Reply:
Now the DirectStoreClient need the clientTelemetry  to initialized first